docs/pages/web/daterange.js

```diff
@@ -0,0 +1,333 @@
+      </MainSection>
+
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Controlled component"
+          description={`
+DateRange is a controlled component. Follow the implementation in the example to implement a controlled DateRange correctly.
+
+When there’s not a date range selected, the call-to-action disables to prevent user errors.
+          `}
+        >
+          <MainSection.Card
+            cardSize="md"
+            sandpackExample={
+              <SandpackExample
+                code={implementation}
+                name="implementation example"
+                previewHeight={500}
+              />
+            }
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="With RadioGroup"
+          description="This variant allow users to select common options of date ranges, so they can select an option without having to navigate through the calendar picker."
+        >
+          <MainSection.Card
+            cardSize="md"
+            title="Future selection"
+            sandpackExample={
+              <SandpackExample
+                code={futureRadiogroup}
+                name="future radiogroup"
+                previewHeight={500}
+              />
+            }
+          />
+          <MainSection.Card
+            cardSize="md"
+            title="Past selection"
+            sandpackExample={
+              <SandpackExample code={pastRadiogroup} name="past radiogroup" previewHeight={500} />
+            }
+          />
+        </MainSection.Subsection>
+
+        <MainSection.Subsection
+          title="Error messaging"
+          description={`
+DateRange can communicate errors when the user selects an invalid date. Use \`startDateErrorMessage\`, \`endDateErrorMessage\` and \`onStartDateChange\`, \`onEndDateChange\`. \`onStartDateBlur\`, \`onStartDateFocus\`, \`onEndDateBlur\`, \`onEndDateFocus\` to implement error messaging correctly.
+
+The following implementation shows how to use all required props for error messaging.
+          `}
+        >
+          <MainSection.Card
+            cardSize="md"
+            sandpackExample={
+              <SandpackExample code={errorMessaging} name="error example" previewHeight={500} />
+            }
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Disable future & past"
+          description="DateField supports disabling future and past dates from being selected."
+        >
+          <MainSection.Card
+            cardSize="md"
+            title="Disable past"
+            sandpackExample={
+              <SandpackExample code={disabledPast} name="past example" previewHeight={500} />
+            }
+          />
+          <MainSection.Card
+            cardSize="md"
+            title="Disable future"
+            sandpackExample={
+              <SandpackExample code={disabledFuture} name="future example" previewHeight={500} />
```

Comment:
I think disabled past and future can be done in the same example with just a fixed range of dates

Reply:
I'm going to keep them separately has there are different use cases for future (select new campaign dates for activation) to past (select previous campaign dates for analytics)